test/command_line/test_export.py

```diff
@@ -405,3 +405,20 @@ def test_json_shortened(dials_data, tmpdir):
     assert d["rlp"][:3] == [-0.5975, -0.6141, 0.4702], d["rlp"][:3]
     assert d["imageset_id"][0] == 0
     assert d["experiment_id"][0] == 0
+
+
+def test_export_sum_or_profile_only(dials_data, tmpdir):
+    expt = dials_data("insulin_processed") / "integrated.expt"
+    refl = dials_data("insulin_processed") / "integrated.refl"
+
+    for remove in "prf", "sum":
+        removed = tmpdir / f"removed_{remove}.refl"
+        data = flex.reflection_table.from_file(refl)
+        del data[f"intensity.{remove}.value"]
+        del data[f"intensity.{remove}.variance"]
+        data.as_file(removed)
+
+        procrunner.run(
+            ["dials.export", expt, removed],
+            working_directory=tmpdir.strpath,
+        )
```

Comment:
Would suggest verifying that `dials.export` actually output an mtz file as expected, rather than simply exiting silently (not tested this change):
```suggestion


def test_export_sum_or_profile_only(dials_data, tmpdir):
    expt = dials_data("insulin_processed") / "integrated.expt"
    refl = dials_data("insulin_processed") / "integrated.refl"

    for remove in "prf", "sum":
        removed = tmpdir / f"removed_{remove}.refl"
        data = flex.reflection_table.from_file(refl)
        del data[f"intensity.{remove}.value"]
        del data[f"intensity.{remove}.variance"]
        data.as_file(removed)

        procrunner.run(
            ["dials.export", expt, removed, f"mtz.hklout=removed_{remove}.mtz"],
            working_directory=tmpdir.strpath,
        )
        assert (tmpdir / "removed_{remove}.mtz").check(file=1)
```
@Anthchirp would probably suggest using the `tmp_path` (returns a `pathlib.Path` object) instead of `tmpdir` (returns a `py.path.local` object) for new tests :)

Reply:
correct, this is because `py.path` is deprecated and `pathlib` is supported out of the box, and therefore it's not really worth learning or propagating the `py.path`-specific features such as `.check(file=1)` - with pathlib it's `.is_file()`